ksql-rest-app/src/test/resources/ksql-plan-schema/schema-final.json

```diff
@@ -0,0 +1,885 @@
+            "$ref" : "#/definitions/DefaultExecutionStepProperties"
+          } ]
+        },
+        "offsetReset" : {
+          "type" : "string",
+          "enum" : [ "EARLIEST", "LATEST" ]
+        },
+        "sourceName" : {
+          "type" : "string"
+        },
+        "alias" : {
+          "type" : "string"
+        }
+      },
+      "title" : "streamSourceV1",
+      "required" : [ "@type", "properties", "offsetReset", "sourceName", "alias" ]
+    },
+    "WindowedStreamSource" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "windowedStreamSourceV1" ],
+          "default" : "windowedStreamSourceV1"
+        },
+        "properties" : {
+          "oneOf" : [ {
+            "$ref" : "#/definitions/DefaultExecutionStepProperties"
+          } ]
+        },
+        "offsetReset" : {
+          "type" : "string",
+          "enum" : [ "EARLIEST", "LATEST" ]
+        },
+        "sourceName" : {
+          "type" : "string"
+        },
+        "alias" : {
+          "type" : "string"
+        }
+      },
+      "title" : "windowedStreamSourceV1",
+      "required" : [ "@type", "properties", "offsetReset", "sourceName", "alias" ]
+    },
+    "StreamStreamJoin" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "streamStreamJoinV1" ],
+          "default" : "streamStreamJoinV1"
+        },
+        "properties" : {
+          "oneOf" : [ {
+            "$ref" : "#/definitions/DefaultExecutionStepProperties"
+          } ]
+        },
+        "joinType" : {
+          "type" : "string",
+          "enum" : [ "INNER", "LEFT", "OUTER" ]
+        },
+        "leftFormats" : {
+          "$ref" : "#/definitions/Formats"
+        },
+        "rightFormats" : {
+          "$ref" : "#/definitions/Formats"
+        },
+        "left" : {
+          "$ref" : "#/definitions/ExecutionStep"
+        },
+        "right" : {
+          "$ref" : "#/definitions/ExecutionStep"
+        },
+        "before" : {
+          "type" : "integer"
+        },
+        "after" : {
+          "type" : "integer"
+        }
+      },
+      "title" : "streamStreamJoinV1",
+      "required" : [ "@type", "properties", "joinType", "leftFormats", "rightFormats", "left", "right", "before", "after" ]
+    },
+    "StreamTableJoin" : {
+      "type" : "object",
+      "additionalProperties" : false,
+      "properties" : {
+        "@type" : {
+          "type" : "string",
+          "enum" : [ "streamTableJoinV1" ],
+          "default" : "streamTableJoinV1"
+        },
+        "properties" : {
+          "oneOf" : [ {
+            "$ref" : "#/definitions/DefaultExecutionStepProperties"
+          } ]
+        },
+        "joinType" : {
+          "type" : "string",
+          "enum" : [ "INNER", "LEFT", "OUTER" ]
```

Comment:
I didn't think `OUTER` isn't allowed for stream-table joins... right?  What about the other join steps?

Reply:
filed https://github.com/confluentinc/ksql/issues/4223